### PR TITLE
refactor: SQL

### LINKED ISSUE #13: Should use the character ID and not the char name
The DB stores the "playerName" but it should be "character_id", it's the proper way to do so

### PR DESCRIPTION
**Pull request by BarbzYHOOL**
I think that was needed, it's clearer now and also

Fixes https://github.com/azerothcore/mod-weapon-visual/issues/13

### CODE REVIEW

File: src/Npc_VisualWeapon.cpp [hunk 1]
```diff
@@ -115,8 +115,8 @@ class npc_visualweapon : public CreatureScript
             itemTemplate->SubClass == ITEM_SUBCLASS_WEAPON_FISHING_POLE)
             return;
 
-        player->SetUInt16Value(PLAYER_VISIBLE_ITEM_1_ENCHANTMENT + (item->GetSlot() * 2), 0, visual);
-        CharacterDatabase.PExecute("REPLACE into custom_item_enchant_visuals VALUES ('%u', '%u', '%s')", item->GetGUIDLow(), visual, player->GetName().c_str());
+        player->SetUInt16Value(PLAYER_VISIBLE_ITEM_1_ENCHANTMENT + (item->GetSlot() * 2), 0, visual_id);
+        CharacterDatabase.PExecute("REPLACE into `mod_weapon_visual__enchantments` VALUES ('%u', '%u', '%s')", item->GetGUIDLow(), visual_id);
```
Comment (Winfidonarleyan) [inline src/Npc_VisualWeapon.cpp]: Doube `_`

**Comment by ghost** [comment]
@BarbzYHOOL 
You know that this will break any existing installation of the module after the upgrade? The old table name has to be changed by the administrator manually.

Concerning the table name I think that "mod_weapon_visual_effect" would be better, as "enchantment" is something different and implies that the weapon gets further stats enhancements or something like that.

File: src/Npc_VisualWeapon.cpp (hunk 1 above)
Comment (BarbzYHOOL) [reply src/Npc_VisualWeapon.cpp]: i thought i had changed that -__- damnit

**Comment by BarbzYHOOL** [comment]
>You know that this will break any existing installation of the module after the upgrade? The old table name has to be changed by the administrator manually.

I know, hmmm the problem is no module make use of "updates/" so I didn't put one but in that case, since it uses item_guid, it should have a convert SQL query, true

**Comment by BarbzYHOOL** [comment]
>Concerning the table name I think that "mod_weapon_visual_effect" would be better, as "enchantment" is something different and implies that the weapon gets further stats enhancements or something like that.

ok and "enchant_id" is right or not?

**Comment by BarbzYHOOL** [comment]
Ok, I could add this I think:

```SQL
ALTER TABLE `custom_item_enchant_visuals`
CHANGE `iguid` `item_guid` int(10) unsigned NOT NULL COMMENT 'item DB guid' FIRST,
CHANGE `display` `enchant_id` int(10) unsigned NOT NULL COMMENT 'enchantID' AFTER `item_guid`,
DROP `PlayerName`,
RENAME TO `mod_weapon_visual_effect`,
COMMENT='Stores the enchant IDs for the visuals of mod-weapon-visual';
```

Where though? in which dir? which name? or maybe just in a readme/issue for now? lol

**Comment by ghost** [comment]
I'd add this to the Readme, perhaps something like "If you use an old version of this module please update the table structure using this SQL statement". As for the column name, probably "enchant_visual_id" would be a more fitting name.

**Comment by ghost** [comment]
Pushed a few updates and alse tested this. Seems to work well, can be merged in my opinion.

**Pull request merged**